### PR TITLE
2 more tests 

### PR DESCRIPTION
**Pull request by sonalk0209**
# Problem Statement
TA for TEST-41279 and TEST-41281
# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/s3/versioning/test_limits_tagging_versioning.py [hunk 1]
```diff
@@ -221,3 +221,155 @@ def test_tag_key_min_41278(self, versioning_status):
...
+        else:
+            s3_cmn_lib.upload_version(self.s3_test_obj, bucket_name=self.bucket_name,
+                                      object_name=self.object_name, file_path=self.file_path,
+                                      versions_dict=self.versions)
+        latest_ver = self.versions[self.object_name]["version_history"][-1]
+        LOGGER.info("Step 2: Successfully uploaded object %s to versioned bucket %s with "
+                    "version ID %s", self.object_name, self.bucket_name, latest_ver)
+        LOGGER.info("Step 3: Perform PUT Object Tagging for %s with 256 char tag value",
+                    self.object_name)
+        resp = s3_cmn_lib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
```
Comment (apoorva-rao) [inline tests/s3/versioning/test_limits_tagging_versioning.py:256]: Let's update the call to pass version_id here, as we are testing the limits when PUT Object Tagging is performed with version_id specified.

File: tests/s3/versioning/test_limits_tagging_versioning.py [hunk 2]
```diff
@@ -221,3 +221,155 @@ def test_tag_key_min_41278(self, versioning_status):
...
+                                             object_name=self.object_name, version_id=latest_ver)
+        assert_utils.assert_true(resp[0], resp)
+        get_tag = resp[1]['TagSet'][0]
+        assert_utils.assert_equal(get_tag, put_tag, "Mismatch in tag Key-Value pair."
+                                                    f"Expected: {put_tag} \n Actual: {get_tag}")
+        LOGGER.info("Step 4: Performed GET Object Tagging for %s with versionId=%s is %s",
+                    self.object_name, latest_ver, get_tag)
+        LOGGER.info("Step 5: Perform PUT Object Tagging for %s with 257 char tag value",
+                    self.object_name)
+        resp = s3_cmn_lib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
```
Comment (apoorva-rao) [inline tests/s3/versioning/test_limits_tagging_versioning.py:279]: Same as the comment above, we need to pass version_id to put_object_tagging

File: tests/s3/versioning/test_limits_tagging_versioning.py [hunk 3]
```diff
@@ -221,3 +221,155 @@ def test_tag_key_min_41278(self, versioning_status):
...
+        else:
+            s3_cmn_lib.upload_version(self.s3_test_obj, bucket_name=self.bucket_name,
+                                      object_name=self.object_name, file_path=self.file_path,
+                                      versions_dict=self.versions)
+        latest_ver = self.versions[self.object_name]["version_history"][-1]
+        LOGGER.info("Step 2: Successfully uploaded object %s to versioned bucket %s with "
+                    "version ID %s", self.object_name, self.bucket_name, latest_ver)
+        LOGGER.info("Step 3: Perform PUT Object Tagging for %s with 10 tag key-value pairs",
+                    self.object_name)
+        resp = s3_cmn_lib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
```
Comment (apoorva-rao) [inline tests/s3/versioning/test_limits_tagging_versioning.py:332]: We would need to specify version_id to the function call

File: tests/s3/versioning/test_limits_tagging_versioning.py [hunk 4]
```diff
@@ -221,3 +221,155 @@ def test_tag_key_min_41278(self, versioning_status):
...
+                                             object_name=self.object_name, version_id=latest_ver)
+        assert_utils.assert_true(resp[0], resp)
+        get_tag = resp[1]['TagSet'][0]
+        assert_utils.assert_equal(get_tag, put_tag, "Mismatch in tag Key-Value pair."
+                                                    f"Expected: {put_tag} \n Actual: {get_tag}")
+        LOGGER.info("Step 4: Performed GET Object Tagging for %s with versionId=%s is %s",
+                    self.object_name, latest_ver, get_tag)
+        LOGGER.info("Step 5: Perform PUT Object Tagging for %s with 11 tag key-value pairs",
+                    self.object_name)
+        resp = s3_cmn_lib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
```
Comment (apoorva-rao) [inline tests/s3/versioning/test_limits_tagging_versioning.py:355]: Same as 332, let's update this to specify version_id

File: tests/s3/versioning/test_limits_tagging_versioning.py [hunk 5]
```diff
@@ -221,3 +221,155 @@ def test_tag_key_min_41278(self, versioning_status):
...
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name, version_tag=self.ver_tag,
+                                             versions_dict=self.versions, tag_count=11)
+        assert_utils.assert_false(resp[0], resp)
+        LOGGER.info("Step 5: PUT Object Tagging for %s with 10 tag key-value pairs failed as "
```
Comment (apoorva-rao) [inline tests/s3/versioning/test_limits_tagging_versioning.py:361]: We can update this log message to "more than 10 tag key-value pairs failed as expected"

**Review by DhobaleVishal — APPROVED** [review]

**Review by apoorva-rao — CHANGES_REQUESTED** [review]
@sonalk0209 minor changes needed, please see the inline comments

**Review by apoorva-rao — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**